test-suite/src/test/scala/org/scalajs/testsuite/library/UnionTypeTest.scala

```diff
@@ -0,0 +1,153 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2015, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.library
+
+import scala.language.implicitConversions
+
+import scala.scalajs.js
+import js.|
+
+import org.scalajs.jasminetest.JasmineTest
+import org.scalajs.testsuite.Typechecking._
+
+import js.JSConverters._
+
+object UnionTypeTest extends JasmineTest {
+
+  private implicit def unionAsJSAny(x: _ | _): js.Any = x.asInstanceOf[js.Any]
```

Comment:
Interpreting "things" as `js.Any` is mostly useful if you work a lot with `js.Dynamic`. Otherwise very few things should be typed as `js.Any`. In this case it's mostly our code because we made Jasmine take `js.Any` to force conversions in typical cases, because most of our test suite is concerned with interop and implementation details of the language.

In more typical code, you have completely typed APIs where you don't have `js.Any`s (you have `Any` instead). If at all, people were using `js.Any` so far where now they would use a union type ^^